### PR TITLE
Upgrade PodOverhead to beta

### PR DESCRIPTION
**Pull request by egernst**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

`PodOverhead` feature is targeting `beta` in 1.18. As part of this graduation, the `PodOverhead` feature-gate will be enabled by default.
 
**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

This PR will be removed from hold once remaining criteria for beta status is completed.

**Does this PR introduce a user-facing change?**:
 
```release-note
NONE
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -614,7 +615,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	LocalStorageCapacityIsolationFSQuotaMonitoring: {Default: false, PreRelease: featuregate.Alpha},
 	NonPreemptingPriority:                          {Default: false, PreRelease: featuregate.Alpha},
 	VolumePVCDataSource:                            {Default: true, PreRelease: featuregate.Beta},
-	PodOverhead:                                    {Default: false, PreRelease: featuregate.Alpha},
+	PodOverhead:                                    {Default: true, PreRelease: featuregate.Beta},
```
Comment (Huang-Wei) [inline pkg/features/kube_features.go]: Another place needs to be updated as well:

```
	// owner: @egernst
	// alpha: v1.16
	// <-- add a line indicating it's beta in 1.18
	// Enables PodOverhead, for accounting pod overheads which are specific to a given RuntimeClass
	PodOverhead featuregate.Feature = "PodOverhead"
```

Comment (egernst) [reply pkg/features/kube_features.go]: Can you clarify?I *think* this is already there?

Comment (Huang-Wei) [reply pkg/features/kube_features.go]: nvm... Not sure whey I didn't see that...

**Pull request merged**